cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/Operator.java

```diff
@@ -81,6 +83,12 @@ default void reconcileThese(String trigger, Set<NamespaceAndName> desiredNames,
             metrics().pausedResourceCounter(namespace).set(0);
         }
 
+        for (NamespaceAndName desiredName : desiredNames) {
+            Tags tags = metrics().getTags(desiredName.getName(), namespace, kind());
+            metrics().removeMetric(MetricsHolder.METRICS_SERVER_CERTIFICATE_EXPIRATION_MS, tags);
+            metrics().removeMetric(MetricsHolder.METRICS_CLIENT_CERTIFICATE_EXPIRATION_MS, tags);
+        }
```

Comment:
Doesn't this remove the metrics for the existing clusters instead of removing metrics for clusters that do not exist anymore?

---

Also, should we call this only for the Kafka kind?

Reply:
We're in the middle of testing currently 😄  But you might be right, we will get there soon, now creation has been tested through.